scripts/o3de/ExportScripts/export_source_android.py

```diff
@@ -0,0 +1,383 @@
+    
+    if is_installer_sdk:
+        # Check if we have any monolithic entries if building monolithic
+        if not exp.has_monolithic_artifacts(ctx):
+            logger.error("Trying to create monolithic build, but no monolithic artifacts are detected in the engine installation!.")
+            raise exp.ExportProjectError("Trying to build monolithic without libraries.")
+        tools_build_path = engine_path / 'bin' / exp.get_platform_installer_folder_name() / tool_config / 'Default'
+    
+    if not tools_build_path:
+        tools_build_path = default_base_path / 'build/tools'
+    elif not tools_build_path.is_absolute():
+        tools_build_path = default_base_path / tools_build_path
+    
+    logger.info(f"Tools build path set to {tools_build_path}")
+    
+    if should_build_tools and not is_installer_sdk:
+        exp.build_export_toolchain(ctx=ctx,
+                                   tools_build_path=tools_build_path,
+                                   engine_centric=engine_centric,
+                                   tool_config=tool_config,
+                                   logger=logger)
+
+    if should_build_all_assets:
+        asset_processor_path = exp.get_asset_processor_batch_path(tools_build_path=tools_build_path,
+                                                                  using_installer_sdk=is_installer_sdk,
+                                                                  tool_config=tool_config,
+                                                                  required=True)
+        logger.info(f"Using '{asset_processor_path}' to process the assets.")
+        exp.build_assets(ctx=ctx,
+                         tools_build_path=tools_build_path,
+                         engine_centric=engine_centric,
+                         fail_on_ap_errors=fail_on_asset_errors,
+                         using_installer_sdk=is_installer_sdk,
+                         tool_config=tool_config,
+                         selected_platform=expected_platform,
+                         logger=logger)
+
+    asset_bundles_path = exp.bundle_assets(ctx=ctx,
+                      selected_platform=expected_platform,
+                      seedlist_paths=validated_seedslist_paths,
+                      seedfile_paths=preprocessed_seedfile_paths,
+                      tools_build_path=tools_build_path,
+                      engine_centric=engine_centric,
+                      asset_bundling_path=project_path/'AssetBundling',
+                      using_installer_sdk=is_installer_sdk,
+                      tool_config=tool_config,
+                      max_bundle_size=max_bundle_size)
+
+    android_project_config= android_support.get_android_config(project_path=None)
+
+    android_project_config.set_config_value(key=android_support.SETTINGS_ASSET_BUNDLE_SUBPATH.key, value=str(asset_bundles_path).replace('\\', '/'), validate_value=False)
+    
```

Comment:
The configs are designed to set default values and should only be set explicitly by the user. It looks like we are missing the `SETTINGS_ASSET_BUNDLE_SUBPATH` as an override parameter to the [add_args](https://github.com/o3de/o3de/blob/8fc008743313813933cd0be190535fc4ed79a92b/scripts/o3de/o3de/android.py#L517-L519) for the `android-generate` command. Remove this call but add an override argument to that method instead.